autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/load/YarnLoadEvaluator.java

```diff
@@ -118,74 +125,105 @@ protected void execute() {
+
+        YarnScalingServiceV1Response yarnResponse =
+                yarnMetricsClient.getYarnMetricsForCluster(cluster, stackV4Response, policyHostGroup, pollingUserCrn, mandatoryDownScaleCount);
 
         if (cluster.getUpdateFailedDetails() != null && pollingUserCrn.equals(cluster.getMachineUserCrn())) {
             // Successful YARN API call
             clusterService.setUpdateFailedDetails(cluster.getId(), null);
         }
 
-        int yarnRecommendedScaleUpCount = yarnResponseUtils.getYarnRecommendedScaleUpCount(yarnResponse, policyHostGroup,
-                maxAllowedUpScale, mandatoryUpScaleCount, loadAlertConfiguration.getMaxScaleUpStepSize());
-        List<String> yarnRecommendedDecommissionHosts = yarnResponseUtils.
-                getYarnRecommendedDecommissionHostsForHostGroup(cluster.getStackCrn(), yarnResponse,
-                        hostFqdnsToInstanceId, maxAllowedDownScale, mandatoryDownScaleCount, loadAlertConfiguration.getMaxScaleDownStepSize());
+        int yarnRecommendedScaleUpCount = yarnResponseUtils.getYarnRecommendedScaleUpCount(yarnResponse, policyHostGroup, maxAllowedUpScale,
+                mandatoryUpScaleCount, loadAlertConfiguration.getMaxScaleUpStepSize(), mandatoryStopStartUpscaleCount);
+        List<String> yarnRecommendedDecommissionHosts =
+                yarnResponseUtils.getYarnRecommendedDecommissionHostsForHostGroup(yarnResponse, hostFqdnsToInstanceId, stoppedHostInstanceIds,
+                        maxAllowedDownScale, mandatoryDownScaleCount, mandatoryStopStartDownscaleCount, loadAlertConfiguration.getMaxScaleDownStepSize());
 
-        LOGGER.info("yarnRecommendedScaleUpCount={}, yarnRecommendedDecommssion={}", yarnRecommendedScaleUpCount, yarnRecommendedDecommissionHosts);
+        LOGGER.info("yarnRecommendedScaleUpCount={}, yarnRecommendedDecommission={}", yarnRecommendedScaleUpCount, yarnRecommendedDecommissionHosts);
+
+        ScalingAdjustmentType adjustmentType = !isStopStartEnabled || mandatoryStopStartUpscaleCount.isPresent() || mandatoryStopStartDownscaleCount.isPresent()
+                ? REGULAR : STOPSTART;
 
         fireAppropriateEvent(yarnRecommendedScaleUpCount, stackV4Response, existingHostGroupSize, serviceHealthyHostGroupSize,
-                yarnRecommendedDecommissionHosts);
+                yarnRecommendedDecommissionHosts, stoppedHostInstanceIds, adjustmentType);
     }
 
     private void fireAppropriateEvent(int yarnRecommendedScaleUpCount, StackV4Response stackV4Response, int existingHostGroupSize,
-            int serviceHealthyHostGroupSize, List<String> yarnRecommendedDecommissionHosts) {
+            int serviceHealthyHostGroupSize, List<String> yarnRecommendedDecommissionHosts, List<String> stoppedHosts, ScalingAdjustmentType adjustmentType) {
         if (yarnRecommendedScaleUpCount > 0 && isCoolDownTimeElapsed(cluster.getStackCrn(), "scaled-up",
```

Comment:
Unrelated to the patch: Is the isCooldownTimeElapsed check needed again? (Guessing it's here because we don'e necessarily know whether a scale operation is an upscale or downscale when we start the processing)

Reply:
From the code, coolDownTime is explcity checked for `scaleUp`,`scaleDown` and `polled` separately. I have respected this check in this refactored patch